pkg/admin/admin.go

```diff
@@ -28,6 +28,7 @@ func StartServer(addr string) {
 		Handler:      h,
 		ReadTimeout:  10 * time.Second,
 		WriteTimeout: 10 * time.Second,
+		IdleTimeout:  1 * time.Minute,
```

Comment:
Can we just unset all of these timeouts? Why are they set?

Reply:
If we unset these, we could simplify this all to `http.ListenAndServe(addr, h)`

Reply:
Here's where it was introduced https://github.com/linkerd/linkerd2/commit/682b0274b5ff4535d1c4d088650cf6d13e508b31#diff-b5ab530c184ee6c6d6fe09ad488de24aL79 -- previously we used the simpler constructor. No explanation in commit message or code.

Reply:
read and write timeouts seem like reasonable safety nets for these unary APIs.  If any HTTP message takes longer than 10 seconds to read or write, we probably want to abort it.

Removing idle timeout would be fine, I think, but it doesn't appear to be possible to have a read timeout but no idle timeout (????) since an unset idle timeout defaults to the read timeout.  But as long as this is long enough and we have the liveness checks to keep the connection open, this should amount to the same thing.

Reply:
As I mentioned in the issue, I think we could set a ReadHeaderTimeout if we really want to, but I would prefer just using the defaults. In all of the cases we care about (prometheus, healthchecking), the clients will enforce some timeout.

If we do want to keep timeouts here, we need to document why we've chosen these values and what problem they solve. I don't want to cargo-cult arbitrary configuration (that has caused problems).

I'm specifically -1 on setting an IdleTimeout, though. We can clients choose how they want to manage connections.